staging/src/k8s.io/client-go/dynamic/golden_test.go

```diff
@@ -338,7 +339,10 @@ func TestGoldenResponse(t *testing.T) {
 
 				var got []interface{}
 				for e := range w.ResultChan() {
-					u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&e)
```

Comment:
This was marshaling watch.Event types to json, which are not serializable

Fixing ToUnstructured to match stdlib changed the case of untagged struct fields.

This change serializes the actual wire metav1.WatchEvent types